scripts/build/check-dependency.sh

```diff
@@ -0,0 +1,32 @@
+#!/bin/bash
+
+###
+# Copyright 2016 resin.io
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+###
+
+set -e
+set +u
+ARGV_DEPENDENCY=$1
+set -u
+
+if [ -z "$ARGV_DEPENDENCY" ]; then
+  echo "Usage: $0 <dependency>"
+  exit 1
+fi
+
+if ! command -v $ARGV_DEPENDENCY 2>/dev/null 1>&2; then
+  echo "Dependency missing: $ARGV_DEPENDENCY" 1>&2
+  exit 1
+fi
```

Comment:
I wonder if you could extend this to optionally take a second (or more?) argument(s), and then return the first of those arguments that actually exists? I.e. the stuff you do in `download-tool.sh` to work out which SHA256SUM command to use.